### PR TITLE
Use WebAppManifest, WebIDL, etc.

### PR DESCRIPTION
**Pull request by plehegar**
This is intended for discussion with regards to integrating with the Web platform while remaining true to MiniApp.

### CODE REVIEW

**Comment by xfq** [comment]
* Preview: https://raw.githack.com/w3c/miniapp/musing-on-webappmanifest/specs/manifest/index.html

* Diff: https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fw3c.github.io%2Fminiapp%2Fspecs%2Fmanifest%2F&doc2=https%3A%2F%2Fraw.githack.com%2Fw3c%2Fminiapp%2Fmusing-on-webappmanifest%2Fspecs%2Fmanifest%2Findex.html

-----

Comparison of the two formats (before the PR): https://github.com/w3c/miniapp/blob/gh-pages/specs/manifest/docs/explainer.md#4-relationship-with-web-app-manifest

File: specs/manifest/index.html [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
     <script class="remove">
         var respecConfig = {
-            specStatus: "CG-DRAFT",
+            specStatus: "ED",
```
Comment (xfq) [inline specs/manifest/index.html:33]: This should probably be `CG-DRAFT` per https://github.com/w3c/respec/issues/3017

File: specs/manifest/index.html [hunk 2]
```diff
@@ -297,161 +308,210 @@ <h2>Manifest</h2>
       </tfoot>
     </table>
 
+    <p class="note">
```
Comment (xfq) [inline specs/manifest/index.html:311]: Note that some of the members are being (have been?) moved to https://w3c.github.io/manifest-app-info/

File: specs/manifest/index.html (hunk 1 above)
Comment (plehegar) [reply specs/manifest/index.html]: Fiex

File: specs/manifest/index.html (hunk 2 above)
Comment (plehegar) [reply specs/manifest/index.html:319]: Fixed. Now using the new spec as well.

**Comment by plehegar** [comment]
@marcoscaceres and @aarongustafson , this is an attempt to better align MiniApp with the Web App Manifest specifications.

File: specs/manifest/index.html [hunk 3]
```diff
@@ -57,14 +57,32 @@
...
+                href: 'https://semver.org/'
+              },
+             'manifest-app-info': {
+                title: 'Web App Manifest - Application Information',
+                href: 'https://w3c.github.io/manifest-app-info/',
+                authors: [
+                  'Aaron Gustafson'
+                ],
+                date: 'July 2020'
+              }
```
Comment (marcoscaceres) [inline specs/manifest/index.html:72]: We will publish on TR in a week :) 

```suggestion
```

File: specs/manifest/index.html [hunk 4]
```diff
@@ -297,161 +316,236 @@ <h2>Manifest</h2>
...
+      {{WebAppManifest/shortcuts}}) are not currently supported by MiniApp user agents.
+    </p>
+
     <section>
       <h2>
-        <span its-locale-filter-list="en" lang="en">Dictionary</span>
+        <span its-locale-filter-list="en" lang="en"><dfn>MiniAppManifest</dfn> Dictionary</span>
         <span its-locale-filter-list="zh-hans" lang="zh-hans">字段</span>
       </h2>
+      <pre class="idl">
```
Comment (marcoscaceres) [inline specs/manifest/index.html:333]: Please avoid doing this. We are removing WebIDL form Web Manifest also. Web IDL is not suitable for JSON.

**Comment by marcoscaceres** [comment]
Please drop all the IDL stuff. Turned out to be a terrible idea for Web Manifest - don't repeat our mistake.

File: specs/manifest/index.html (hunk 4 above)
Comment (plehegar) [reply specs/manifest/index.html:333]: oh, just when I was thinking the opposite.  I found it useful in specifying the constraints, such as required members, valid values, and default values. Are you folks moving to JSON Schema or something else?

**Comment by plehegar** [comment]
I don't feel strongly about the WebIDL, so happy to do so. It's more important to be aligned with what Web App Manifest does in my mind.

File: specs/manifest/index.html (hunk 4 above)
Comment (aarongustafson) [reply specs/manifest/index.html:333]: Yes, JSON Schema converting to Infra (see https://github.com/w3c/manifest/issues/611#issuecomment-491675944).

Progress to that end: https://github.com/w3c/manifest/pull/928

Comment (espinr) [reply specs/manifest/index.html:333]: +1 to use Infra, but WebIDL could be kept as non-normative. It's more human-readable at a glance.

**Pull request closed without merging**